File: Test1W.Java/Test1W.java
class Test1W{
	public static void main(String args[]) {
        //System.out.println("begin");
        long startTime = System.currentTimeMillis();
        int total=0;
        for (int i = 0; i < 10000; i++) {
            total+=NarcissisticNumber();
        }
        long endTime = System.currentTimeMillis();
        System.out.println("time:" + (endTime - startTime)+",total:"+total );
    }

    public static int NarcissisticNumber() {
        int count=0;
        int ge, shi, bai;
        for (int n = 100; n < 100000; n++) {
            ge = n % 10;
            shi = (n - ge) / 10 % 10;
            bai = (int) (n / 100);
            if (ge * ge * ge + shi * shi * shi + bai * bai * bai == n) {
                count++;
                    //System.out.println(String.valueOf(n));
            }
        }
        return count;
	}
 }

File: Test1W.Go/Test1W.go
package main
 
import (
    "fmt"
    "time"
)

func NarcissisticNumber() int {
    var count int =0;
    for n := 100; n < 100000; n++ {
        ge := n % 10;
        shi := (n - ge) / 10 % 10;
        bai := n / 100;
        if (ge * ge * ge + shi * shi * shi + bai * bai * bai == n) {
            count++;
            //fmt.Printf("%d",n)
        }
    }
    return count;
}

func main() {
    var total int =0;
    var startTime = time.Now();//获取当前时间
    for  i := 0; i < 10000; i++ {
        total += NarcissisticNumber();
    }
    //var endTime = time.Now();
    elapsed := time.Since(startTime)
    //fmt.Printf("%s%d%s%d","time:",endTime.Sub(startTime), "ms,total:", total);
    fmt.Printf("%s%d%s%d","time:",elapsed/1000/1000, "ms,total:", total);
}
